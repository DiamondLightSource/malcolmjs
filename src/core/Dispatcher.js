/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
