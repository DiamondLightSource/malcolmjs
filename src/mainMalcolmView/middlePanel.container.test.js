import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import configureStore from 'redux-mock-store';
import MiddlePanel from './middlePanel.container';

describe('MalcolmMiddlePanel', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('renders correctly', () => {
    const state = {
      malcolm: {
        parentBlock: 'test1',
        mainAttribute: 'layout',
        blocks: {
          test1: {
            name: 'Test main view',
            attributes: [
              {
                name: 'layout',
                layout: {
                  blocks: [
                    {
                      name: 'counter 1',
                      mri: 'block2',
                      visible: 'true',
                      position: {
                        x: 0,
                        y: 0,
                      },
                    },
                  ],
                },
              },
            ],
          },
          block2: {
            name: 'block2',
            attributes: [
              {
                name: 'in1',
                mri: 'in1',
                meta: {
                  tags: ['inport:bool'],
                },
              },
              {
                name: 'out1',
                mri: 'out1',
                meta: {
                  tags: ['outport:bool'],
                },
              },
              {
                name: 'icon',
                mri: 'icon',
                value: 'icon svg',
                meta: {
                  tags: ['widget:icon'],
                },
              },
            ],
          },
        },
      },
    };

    const mockStore = configureStore();

    const wrapper = shallow(<MiddlePanel store={mockStore(state)} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders a container only if the mainAttribute is not layout', () => {
    const state = {
      malcolm: {
        mainAttribute: 'health',
        blocks: {},
      },
    };

    const mockStore = configureStore();

    const wrapper = shallow(<MiddlePanel store={mockStore(state)} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
