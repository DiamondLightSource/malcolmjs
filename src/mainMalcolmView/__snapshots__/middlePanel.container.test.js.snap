// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MalcolmMiddlePanel renders a container only if the no widget tag found for mainAttribute 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-plainBackground-8"
  />
</div>
`;

exports[`MalcolmMiddlePanel renders alarmed correctly 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-layoutArea-2"
  >
    <Connect(Layout) />
    <div
      className="MiddlePanelContainer-alarm-3"
      style={
        Object {
          "bottom": 12,
          "left": 389,
        }
      }
    >
      <WithTheme(AttributeAlarm)
        alarmSeverity={4}
      />
    </div>
    <div
      className="MiddlePanelContainer-paletteButton-5"
      style={
        Object {
          "bottom": 32,
          "right": 32,
        }
      }
    >
      <WithStyles(Button)
        color="secondary"
        data-cy="palettebutton"
        onClick={[Function]}
        variant="fab"
      >
        <pure(PaletteIcon) />
      </WithStyles(Button)>
    </div>
    <div
      className="MiddlePanelContainer-autoLayoutButton-6"
      style={
        Object {
          "right": 29,
          "top": 12,
        }
      }
    >
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
        variant="raised"
      >
        Auto layout
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`MalcolmMiddlePanel renders an archive viewer pane if the mainAttribute is a "standard" attribute allowed type 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-plainBackground-8"
  >
    <div
      className="MiddlePanelContainer-tableContainer-7"
      style={
        Object {
          "left": 360,
          "width": "calc(100% - 360px)",
        }
      }
    >
      <Connect(WithTheme(AttributeViewer))
        attributeName="health"
        blockName="test1"
        typeId="malcolm:core/StringMeta:1.0"
        widgetTag="widget:textupdate"
      />
    </div>
  </div>
</div>
`;

exports[`MalcolmMiddlePanel renders errored correctly 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-layoutArea-2"
  >
    <Connect(Layout) />
    <div
      className="MiddlePanelContainer-alarm-3"
      style={
        Object {
          "bottom": 12,
          "left": 389,
        }
      }
    >
      <WithTheme(AttributeAlarm)
        alarmSeverity={2}
      />
    </div>
    <div
      className="MiddlePanelContainer-paletteButton-5"
      style={
        Object {
          "bottom": 32,
          "right": 32,
        }
      }
    >
      <WithStyles(Button)
        color="secondary"
        data-cy="palettebutton"
        onClick={[Function]}
        variant="fab"
      >
        <pure(PaletteIcon) />
      </WithStyles(Button)>
    </div>
    <div
      className="MiddlePanelContainer-autoLayoutButton-6"
      style={
        Object {
          "right": 29,
          "top": 12,
        }
      }
    >
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
        variant="raised"
      >
        Auto layout
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`MalcolmMiddlePanel renders layout correctly 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-layoutArea-2"
  >
    <Connect(Layout) />
    <div
      className="MiddlePanelContainer-alarm-3"
      style={
        Object {
          "bottom": 12,
          "left": 389,
        }
      }
    >
      <WithTheme(AttributeAlarm)
        alarmSeverity={0}
      />
    </div>
    <div
      className="MiddlePanelContainer-paletteButton-5"
      style={
        Object {
          "bottom": 32,
          "right": 32,
        }
      }
    >
      <WithStyles(Button)
        color="secondary"
        data-cy="palettebutton"
        onClick={[Function]}
        variant="fab"
      >
        <pure(PaletteIcon) />
      </WithStyles(Button)>
    </div>
    <div
      className="MiddlePanelContainer-autoLayoutButton-6"
      style={
        Object {
          "right": 29,
          "top": 12,
        }
      }
    >
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
        variant="raised"
      >
        Auto layout
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`MalcolmMiddlePanel renders pending correctly 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-layoutArea-2"
  >
    <Connect(Layout) />
    <div
      className="MiddlePanelContainer-alarm-3"
      style={
        Object {
          "bottom": 12,
          "left": 389,
        }
      }
    >
      <WithTheme(AttributeAlarm)
        alarmSeverity={-1}
      />
    </div>
    <div
      className="MiddlePanelContainer-paletteButton-5"
      style={
        Object {
          "bottom": 32,
          "right": 32,
        }
      }
    >
      <WithStyles(Button)
        color="secondary"
        data-cy="palettebutton"
        onClick={[Function]}
        variant="fab"
      >
        <pure(PaletteIcon) />
      </WithStyles(Button)>
    </div>
    <div
      className="MiddlePanelContainer-autoLayoutButton-6"
      style={
        Object {
          "right": 29,
          "top": 12,
        }
      }
    >
      <WithStyles(Button)
        color="primary"
        disabled={true}
        onClick={[Function]}
        variant="raised"
      >
        Auto layout
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`MalcolmMiddlePanel renders table correctly 1`] = `
<div
  className="MiddlePanelContainer-container-1"
  role="presentation"
>
  <div
    className="MiddlePanelContainer-plainBackground-8"
  >
    <div
      className="MiddlePanelContainer-tableContainer-7"
      style={
        Object {
          "left": 360,
          "width": "calc(100% - 360px)",
        }
      }
    >
      <Connect(WithTheme(TableContainer))
        attributeName="layout"
        blockName="test1"
        footerItems={
          Array [
            <WithStyles(Tooltip)
              id="1"
              placement="right"
              title=""
          >
              <WithStyles(IconButton)
                  className="MiddlePanelContainer-button-9"
                  disableRipple={true}
              >
                  <WithTheme(AttributeAlarm)
                      alarmSeverity={0}
                  />
              </WithStyles(IconButton)>
          </WithStyles(Tooltip)>,
          ]
        }
      />
    </div>
  </div>
</div>
`;
